lib/web_ui/dev/webdriver_browser.dart

```diff
@@ -0,0 +1,96 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+import 'dart:math';
+
+import 'package:image/image.dart';
+import 'package:webdriver/io.dart' show createDriver, WebDriver;
+
+import 'browser.dart';
+
+abstract class WebDriverBrowserEnvironment extends BrowserEnvironment {
+  late final Process _driverProcess;
+
+  Future<Process> spawnDriverProcess();
+  Uri get driverUri;
+
+  @override
+  Future<void> prepare() async {
```

Comment:
Sometimes there's an existing driver process hanging around on the system. It may be useful to check that the port is not already taken. Otherwise, the test might end up using the wrong driver process (depending on when exactly the newly spawned process crashes due to the taken port).

Reply:
I found a nice simple implementation of a "find unused port" function in the dart SDK, which I am shamefully stealing here. That should solve this problem.